hadoop-hdds/test-utils/src/main/java/org/apache/ozone/test/GenericTestUtils.java

```diff
@@ -468,4 +470,48 @@ public static String anyHostWithFreePort() {
     }
   }
 
+  /**
+   * This class is a utility class for java reflection operations.
+   */
+  public static final class ReflectionUtils {
+
+    /**
+     * This method provides the modifiers field using reflection approach which is compatible
+     * for both pre Java 9 and post java 9 versions.
+     * @return modifiers field
+     * @throws IllegalAccessException
+     * @throws NoSuchFieldException
+     */
+    public static Field getModifiersField() throws IllegalAccessException, NoSuchFieldException {
+      Field modifiersField = null;
+      try {
+        modifiersField = Field.class.getDeclaredField("modifiers");
+      } catch (NoSuchFieldException e) {
+        try {
+          Method getDeclaredFields0 = Class.class.getDeclaredMethod(
+              "getDeclaredFields0", boolean.class);
+          boolean accessibleBeforeSet = getDeclaredFields0.isAccessible();
+          getDeclaredFields0.setAccessible(true);
+          Field[] fields = (Field[]) getDeclaredFields0.invoke(Field.class, false);
+          getDeclaredFields0.setAccessible(accessibleBeforeSet);
+          for (Field field : fields) {
+            if ("modifiers".equals(field.getName())) {
+              modifiersField = field;
+              break;
+            }
+          }
+          if (modifiersField == null) {
+            throw e;
+          }
+        } catch (NoSuchMethodException ex) {
+          e.addSuppressed(ex);
+          throw e;
+        } catch (InvocationTargetException ex) {
+          e.addSuppressed(ex);
+          throw e;
+        }
```

Comment:
nit: can combine identical `catch` blocks?

```suggestion
        } catch (InvocationTargetException | NoSuchMethodException ex) {
          e.addSuppressed(ex);
          throw e;
        }
```